internal/service/ocm/ocm_impl.go

```diff
@@ -0,0 +1,147 @@
+package ocm
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	v1 "github.com/openshift-online/ocm-sdk-go/accountsmgmt/v1"
+	"github.com/openshift-online/ocm-sdk-go/logging"
+	"github.com/redhatinsights/mbop/internal/models"
+)
+
+type SDK struct {
+	client *sdk.Connection
+}
+
+func (ocm *SDK) InitSdkConnection(ctx context.Context) error {
+	// Create a logger that has the debug level enabled:
+	logger, err := logging.NewGoLoggerBuilder().
+		Debug(true).
+		Build()
+
+	if err != nil {
+		return err
+	}
+
+	ocm.client, err = sdk.NewConnectionBuilder().
+		Logger(logger).
+
+		// SA Auth:
+		Client(os.Getenv("COGNITO_APP_CLIENT_ID"), os.Getenv("COGNITO_APP_CLIENT_SECRET")).
+
+		// Offline Token Auth:
+		// Tokens(<token>).
+
+		// Oauth Token URL:
+		TokenURL(os.Getenv("OAUTH_TOKEN_URL")).
+
+		// Route to hit for AMS:
+		URL(os.Getenv("AMS_URL")).
+
+		// SA Scopes:
+		Scopes(os.Getenv("COGNITO_SCOPE")).
+		BuildContext(ctx)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ocm *SDK) GetUsers(usernames models.UserBody, q models.UserQuery) (models.Users, error) {
+	search := createSearchString(usernames)
+	collection := ocm.client.AccountsMgmt().V1().Accounts().List().Search(search)
+
+	collection = collection.Order(createQueryOrder(q))
+
+	users := models.Users{}
+	usersResponse, err := collection.Send()
+
+	if err != nil {
+		return users, err
+	}
+
+	if usersResponse.Items().Empty() {
+		return users, err
+	}
+
+	users = responseToUsers(usersResponse)
+
+	return users, err
+}
+
+func (ocm *SDK) IsOrgAdmin(id string) (bool, error) {
+	search := fmt.Sprintf("account.id='%s' and role.id='OrganizationAdmin'", id)
+
+	collection := ocm.client.AccountsMgmt().V1().RoleBindings()
```

Comment:
Since this is currently called on each user in the request, we may run into potential performance issues/n+1 request creep depending on the number of users. Would it be possible/make sense to get role bindings for all users (assuming the data can be grouped/ordered by user) and then map those roles to the user when you iterate over them here: https://github.com/RedHatInsights/mbop/pull/35/files#diff-0b8aed84f503a387e2214aafb0bd26990d42acd1a4f80f9c2679f5fcc3a04289R64 vs making a call for roles per user?